File: src/main/java/gargoyle/ct/pref/impl/prop/CTPrefDoubleProperty.java
package gargoyle.ct.pref.impl.prop;

import gargoyle.ct.pref.CTPreferencesProvider;
import gargoyle.ct.prop.CTNumberProperty;
import org.jetbrains.annotations.NotNull;

public class CTPrefDoubleProperty extends CTPrefProperty<Double> implements CTNumberProperty<Double> {
    public CTPrefDoubleProperty(@NotNull CTPreferencesProvider provider, String name) {
        this(provider, name, 0);
    }

    public CTPrefDoubleProperty(@NotNull CTPreferencesProvider provider, String name, double def) {
        super(Double.class, provider, name, def);
    }
}

File: src/main/kotlin/gargoyle/ct/pref/impl/prop/CTPrefDoubleProperty.kt
package gargoyle.ct.pref.impl.prop

import gargoyle.ct.pref.CTPreferencesProvider
import gargoyle.ct.prop.CTNumberProperty

class CTPrefDoubleProperty @JvmOverloads constructor(
    provider: CTPreferencesProvider,
    name: String,
    def: Double = 0.0
) : CTPrefProperty<Double>(
    Double::class, provider, name, def
), CTNumberProperty<Double>
